src/shared/components/Posts/components/PostsList/PostsList.tsx

```diff
@@ -0,0 +1,41 @@
+import { FC, useContext } from 'react';
+import { Link } from 'react-router-dom';
+//mui
+import { Card } from '@mui/material';
+//context
+import { PostsContext } from '../../context/PostsContext';
+//components
+import { SinglePost } from '../SinglePost';
+import { Loader } from 'shared/components/Loader';
+//styles
+import styles from './PostsList.module.scss';
+
+const PostsList: FC = ({ children }) => {
+    const { filteredPosts, loading } = useContext(PostsContext);
+
+    const renderMessage = filteredPosts.length === 0 && !loading && (
+        <section className={styles.noPostsMessage}>There is no posts.</section>
+    );
+
+    return (
+        <section className={styles.postsList}>
+            {loading ? (
+                <Loader />
+            ) : (
+                filteredPosts.map((post, index) => (
```

Comment:
maybe the code would be more readable is you put it in rebderFilteredPosts function an call it here